### PR TITLE
Use Gradle Version Catalog

### PR DESCRIPTION
**Pull request by hieuwu**
Use Gradle Version Catalog

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+androidx-lifecycle-runtime-ktx = { module = "androidx.lifecycle:lifecycle-runtime-ktx", version.ref = "lifecycleRuntimeKtx" }
+androidx-room-compiler = { module = "androidx.room:room-compiler", version.ref = "roomRuntime" }
+androidx-room-ktx = { module = "androidx.room:room-ktx", version.ref = "roomRuntime" }
+androidx-room-runtime = { module = "androidx.room:room-runtime", version.ref = "roomRuntime" }
+androidx-ui = { module = "androidx.compose.ui:ui", version.ref = "uiToolingPreview" }
+androidx-ui-test-junit4 = { module = "androidx.compose.ui:ui-test-junit4", version.ref = "uiToolingPreview" }
+androidx-ui-test-manifest = { module = "androidx.compose.ui:ui-test-manifest", version.ref = "uiToolingPreview" }
+androidx-ui-tooling = { module = "androidx.compose.ui:ui-tooling", version.ref = "uiToolingPreview" }
+androidx-work-runtime-ktx = { module = "androidx.work:work-runtime-ktx", version.ref = "workRuntimeKtx" }
+animations-core = { module = "io.github.raamcosta.compose-destinations:animations-core", version.ref = "animationsCore" }
```
Comment (mdrlzy) [inline gradle/libs.versions.toml:46]: Maybe `compose-destinations`?
This is lib for navigation
Rename version ref too

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+converter-gson = { module = "com.squareup.retrofit2:converter-gson", version.ref = "converterGson" }
+dagger = { module = "com.google.dagger:dagger", version.ref = "dagger" }
+dagger-compiler = { module = "com.google.dagger:dagger-compiler", version.ref = "dagger" }
+
+firebase-bom = { module = "com.google.firebase:firebase-bom", version.ref = "firebaseBom" }
+firebase-analytics = { module = "com.google.firebase:firebase-analytics-ktx", version = "" }
+firebase-crashlytics = { module = "com.google.firebase:firebase-crashlytics-ktx", version = "" }
+
+junit = { module = "junit:junit", version.ref = "junit" }
+ksp = { module = "io.github.raamcosta.compose-destinations:ksp", version.ref = "animationsCore" }
```
Comment (mdrlzy) [inline gradle/libs.versions.toml:59]: `ksp` sounds unclear
maybe `compose-destinations-compiler`

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+qrgenerator = "1.0.1"
+retrofit = "2.11.0"
+roomRuntime = "2.6.1"
+tagcloud = "1.1.0"
+timber = "5.0.1"
+uiToolingPreview = "1.6.8"
+workRuntimeKtx = "2.8.1"
+
+[libraries]
+about = { module = "dev.arkbuilders.components:about", version.ref = "about" }
```
Comment (mdrlzy) [inline gradle/libs.versions.toml:31]: `ark-about`?

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+androidx-activity-compose = { module = "androidx.activity:activity-compose", version.ref = "activityCompose" }
+androidx-datastore-preferences = { module = "androidx.datastore:datastore-preferences", version.ref = "datastorePreferences" }
+androidx-espresso-core = { module = "androidx.test.espresso:espresso-core", version.ref = "espressoCore" }
+androidx-glance-appwidget = { module = "androidx.glance:glance-appwidget", version.ref = "glanceAppwidget" }
+androidx-junit = { module = "androidx.test.ext:junit", version.ref = "junitVersion" }
+androidx-lifecycle-runtime-ktx = { module = "androidx.lifecycle:lifecycle-runtime-ktx", version.ref = "lifecycleRuntimeKtx" }
+androidx-room-compiler = { module = "androidx.room:room-compiler", version.ref = "roomRuntime" }
+androidx-room-ktx = { module = "androidx.room:room-ktx", version.ref = "roomRuntime" }
+androidx-room-runtime = { module = "androidx.room:room-runtime", version.ref = "roomRuntime" }
+androidx-ui = { module = "androidx.compose.ui:ui", version.ref = "uiToolingPreview" }
```
Comment (mdrlzy) [inline gradle/libs.versions.toml:41]: version ref looks weird
maybe `composeUI`?

File: gradle/libs.versions.toml (hunk 1 above)
Comment (hieuwu) [reply gradle/libs.versions.toml]: Updated

**Pull request merged**